### ISSUE TITLE
Babel can't transform unparanthesized `async` arrows anymore

### ISSUE THREAD

**Issue by inikulin** [babel/babel#4604]
### Input Code

```js
test('foo', async t => await bar());
```

### Babel Configuration (.bablerc, package.json, cli command)

```js
{
  "compact": false,
  "presets": [
    ["es2015", { "loose": true }],
    "babel-preset-stage-2"
  ]
}
```

### Expected Behavior
Previous version of Babel correctly compiled this code (you can try in playground which uses 6.14.0 currently):
```js
test('foo', (() => {
  var ref = _asyncToGenerator(function* (t) {
    return yield bar();
  });

  return function (_x) {
    return ref.apply(this, arguments);
  };
})());
```

### Current Behavior
With latest version of Babel and preset it compiles to:
```js
test('foo', (() => {
    var _ref = (0, _asyncToGenerator3.default)(function* (t) {
        return await bar();
    });

    return function (_x) {
        return _ref.apply(this, arguments);
    };
})());
```
As you can see `await` is not replaced with `yield` which leads to broken code.

**Comment by danharper** [comment]
6.16.2 was just published. Should be fixed as of #4600. Could you give that a try?

**Comment by al-zah** [comment]
@danharper 
Cant see published 6.16.2 versions on npm, is it only me?

> npm view babel-cli dist-tags
> { latest: '6.16.0' }
> npm view babel-core dist-tags
> { latest: '6.16.0', old: '5.8.38' }

**Comment by hzoo** [comment]
<img width="604" alt="screen shot 2016-09-29 at 9 45 14 am" src="https://cloud.githubusercontent.com/assets/588473/18956446/7427cffc-8629-11e6-81fa-725680b44916.png">

Only the changes packages are updated
